docs/maps-and-sets-usage.md

```diff
@@ -0,0 +1,48 @@
+## Map and Set Usage
+
+You need to wrap Maps and Sets inside an object, and when you want it's update to be reflected (e.g. in React),
+one way you can do it is calling the setState on it:
+
+```js
+import create from "zustand";
+
+const useFooBar = create(() => ({ foo: new Map(), bar: new Set() }));
+
+const { getState: getFooBar, setState: setFooBar } = useFooBar;
+  
+function doSomething() {
+  // Following React's best practices, you should create a new Map/Set when updating them:
+  const foo = new Map(getFooBar().foo);
+  const bar = new Set(getFooBar().bar);
+  
+  // use foo/bar...
+  
+  bar.set("bar");
+  foo.set("foo", "bar");
+  foo.delete("foo");
+  
+  // If you want to update some React component that uses `useFooBar`, you have to call setState
+  // to let React know that an update happened:
+  setFooBar({ foo, set });
```

Comment:
```suggestion
  useFooBar.setState({ foo });
```
Should that ☝️ be fine?

I would actually like this pattern.
```js
useFooBar.setState((prev) => ({ foo: new Map(prev.foo).set("newKey", "newValue") }));
```

Reply:
Now that you said it, I tested here, and indeed it doesnt fail to update :) I should've used like that.